install-deps.sh

```diff
@@ -24,7 +24,9 @@ export LC_ALL=en_US.UTF-8 # the following is vulnerable to i18n
 ARCH=$(uname -m)
 
 function in_jenkins() {
+    set +e
     test -n "$JENKINS_HOME"
+    set -e
```

Comment:
Thank you @epuertat.  Your bash-fu is clearly better than mine.

Reply:
@epuertat I couldn't figure out in which case this function fails. I just tried out random things and found out the error happens from here. Still trying out more random stuffs..

Reply:
I have a doubt that it happens at the final in_jenkins function. 
https://github.com/ceph/ceph/blob/8ef84a4351fc724a50047b1c87283fdc57e97aa3/install-deps.sh#L547

Reply:
@nizamial09 `bash -x` (or `set -x`) will add tons of debugging info. Maybe that helps?

That said, being that the last command in the script I think I get what the issue is:
* Bash implicitly returns the status of the last command executed.
* On a Jenkins node, this will always be `0` (it'd be weird that `echo ".."` failed).
* However, on non-Jenkins nodes, the `in_jenkins` will fail and the overall status of that line will be a failure
* In `src/script/run-make.sh` script we have: `$DRY_RUN source ./install-deps.sh || return 1`

What if we end the `install-deps.sh` script with:
```bash
in_jenkins && echo "CI_DEBUG: End install-deps.sh" || true
```

Reply:
let me try it @epuertat

Reply:
> That said, being that the last command in the script I think I get what the issue is:
> 
> * Bash implicitly returns the status of the last command executed.
> * On a Jenkins node, this will always be `0` (it'd be weird that `echo ".."` failed).
> * However, on non-Jenkins nodes, the `in_jenkins` will fail and the overall status of that line will be a failure
> * In `src/script/run-make.sh` script we have: `$DRY_RUN source ./install-deps.sh || return 1`

You are right @epuertat that actually worked and I think this is the issue. Thanks a lot. I'll make the PR ready for reviews.